src/shared/cloudformation/templateRegistry.ts

```diff
@@ -28,7 +28,7 @@ export class CloudFormationTemplateRegistry extends WatchedFiles<CloudFormation.
 
         let template: CloudFormation.Template | undefined
         try {
-            template = await CloudFormation.load(path)
+            template = await CloudFormation.load(path, false)
```

Comment:
the idea here was to avoid registering random .yaml files which are not CFN/SAM files. Will that still be the case?

Reply:
In the current master branch, when loading cloudformation files it tries to look into the "Resources" field in the YAML and see if serverless function exists and then validate those resources, make sure handlers, codeUri, etc are available before loading in the schema which doesn't really make sense.

Now, it basically just loads the yaml file and then checks if `AWSTemplateFormatVersion` or `Resources` fields exist which is more inline with what was originally expected (from what I understand).

From what I can see all the logic for determining if the file is CFN/SAM is just contained in that one spot

Reply:
Thanks! That explanation is useful in the commit message, so I added it.

In the future, it is worth explaining the rationale in the commit message.